lightning/src/ln/channel.rs

```diff
@@ -8771,6 +8999,29 @@ impl<SP: Deref> InboundV2Channel<SP> where SP::Target: SignerProvider {
 	pub fn get_accept_channel_v2_message(&self) -> msgs::AcceptChannelV2 {
 		self.generate_accept_channel_v2_message()
 	}
+
+	pub fn funding_tx_constructed<L: Deref>(
+		mut self, mut signing_session: InteractiveTxSigningSession, logger: &L
+	) -> Result<(Channel<SP>, msgs::CommitmentSigned, Option<Event>), (Self, ChannelError)>
```

Comment:
If we de-coupled the `internal_funding_tx_constructed` logic from the "convert `Self` to `Channel`" logic (i.e., where the latter is infallible), it would simplify the call sites. You wouldn't need to remove and re-insert an entry in the map for the failure case, IIUC.